src/raftstore/store/peer_storage.rs

```diff
@@ -270,11 +270,28 @@ impl PeerStorage {
     pub fn entries(&self, low: u64, high: u64, max_size: u64) -> raft::Result<Vec<Entry>> {
         try!(self.check_range(low, high));
         let mut ents = Vec::with_capacity((high - low) as usize);
+        if low == high {
+            return Ok(ents);
+        }
         let mut total_size: u64 = 0;
         let mut next_index = low;
         let mut exceeded_max_size = false;
 
         let start_key = keys::raft_log_key(self.get_region_id(), low);
+
+        if low + 1 == high {
+            let handle = self.engine.cf_handle(CF_RAFT).unwrap();
```

Comment:
Not very frequently. When restarted, many inactive region will just commit one empty entry.